lib/Doctrine/DBAL/Schema/Column.php

```diff
@@ -118,9 +119,17 @@ public function setOptions(array $options)
     {
         foreach ($options as $name => $value) {
             $method = "set".$name;
-            if (method_exists($this, $method)) {
-                $this->$method($value);
+            if ( ! method_exists($this, $method)) {
+                // next major: use InvalidArgumentException::fromUnsupportedOption()
+                @trigger_error(sprintf(
+                    'The "%s" option is not supported,'.
+                    ' setting it is deprecated and will cause an exception in 3.0',
+                    $name
+                ), E_USER_DEPRECATED);
+
+                return;
```

Comment:
Why not `return $this` like a few lines after?

Reply:
Good catch, that's a BC-break!